### PR TITLE
Add post uploadDatasetThumbnail endpoint

### PR DESCRIPTION
**Pull request by MYF95**
## Description
Provides the `post uploadDatasetThumbnail` endpoint, where it receives a `datasetId` and an `image` file to update the thumbnail.

## Changes
- Added request-promise module
- Added updateDatasetThumbnail endpoint
- Added tests for the endpoint
  - `it should call request with expected url`
  - `it should call request with expected headers when no apiToken provided`
  - `it should return expected response`
  - `it should throw expected error`

## Tests
- Manual testing the thumbnail updates by using a local image
- Automated test created

## Checklist
- [X] The project builds
- [X] The project passes lint checks
- [X] The project passes format checks
- [X] The project passes unit tests
- [X] I've manually tested the functionality

## Screenshots
<Screenshots of new functionality if available>

### CODE REVIEW

File: src/dataverseClient.ts [hunk 1]
```diff
@@ -84,6 +85,24 @@ export class DataverseClient {
     })
   }
 
+  // TO DO: find a way to implement this endpoint using `axios` instead of `request-promise`
+  // Using `request-promise` was the only way so far to be able to upload an image without any issues
+  public async uploadDatasetThumbnail(datasetId: string, image: object): Promise<any> {
+    const url = `${this.host}/api/datasets/${datasetId}/thumbnail`
+    const options = {
+      url: url,
```
Comment (tainguyenbui) [inline src/dataverseClient.ts]:
```suggestion
      url,
```

File: test/dataverseClient.spec.ts [hunk 2]
```diff
@@ -757,6 +762,76 @@ describe('DataverseClient', () => {
...
+        headers: { 'X-Dataverse-key': apiToken },
+        formData: { file: testImage },
+        resolveWithFullResponse: true
+      }
+
+      requestPostStub.withArgs(expectedRequest).resolves(mockResponse)
+
+      const response = await client.uploadDatasetThumbnail(datasetId, testImage)
+
+      expect(response).to.be.deep.eq(expectedResponse)
```
Comment (tainguyenbui) [inline test/dataverseClient.spec.ts]:
```suggestion
      expect(response).to.be.deep.equal(expectedResponse)
```

File: src/dataverseClient.ts (hunk 1 above)
Comment (MYF95) [reply src/dataverseClient.ts]: Added changes to new commit

File: test/dataverseClient.spec.ts (hunk 2 above)
Comment (MYF95) [reply test/dataverseClient.spec.ts]: Added changes to new commit

**Pull request merged**